File: src/main/java/um/haberes/rest/model/CursoMateria.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import um.haberes.rest.kotlin.model.Curso;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "cursoId", "facultadId",
		"geograficaId", "planId", "carreraId", "materiaId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CursoMateria extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -947881750890493221L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cursoMateriaId;

	private Long cursoId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer planId;
	private Integer carreraId;
	private String materiaId;
	private BigDecimal proporcion = BigDecimal.ZERO;

	@OneToOne
	@JoinColumn(name = "cursoId", insertable = false, updatable = false)
	private Curso curso;

	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;
	
}

File: src/main/java/um/haberes/core/kotlin/model/CursoMateria.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import um.haberes.core.hexagonal.facultad.infrastructure.persistence.entity.FacultadEntity
import um.haberes.core.hexagonal.geografica.infrastructure.persistence.entity.GeograficaEntity
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["cursoId", "facultadId", "geograficaId", "planId", "carreraId", "materiaId"])])
data class CursoMateria(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cursoMateriaId: Long? = null,

    var cursoId: Long? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var planId: Int? = null,
    var carreraId: Int? = null,
    var materiaId: String? = null,
    var proporcion: BigDecimal = BigDecimal.ZERO,

    @OneToOne
    @JoinColumn(name = "cursoId", insertable = false, updatable = false)
    var curso: Curso? = null,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: FacultadEntity? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: GeograficaEntity? = null

) : Auditable()
